src/components/calcite-split-button/calcite-split-button.tsx

```diff
@@ -0,0 +1,145 @@
+
+  /** select theme (light or dark), defaults to light */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of the control, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** specify the icon used for the dropdown menu, defaults to chevron */
+  @Prop({ mutable: true, reflect: true }) dropdownIconType: "chevron" | "caret" = "chevron";
+
+  /** text for primary action button  */
+  @Prop({ reflect: true }) primaryText: string;
+
+  /** optionally pass an icon to display on the primary button - accepts Calcite UI icon names  */
+  @Prop({ reflect: true }) primaryIcon?: string;
+
+  /** optionally pass an aria-label for the primary button */
+  @Prop({ reflect: true }) primaryLabel?: string;
+
+  /** aria label for overflow button */
+  @Prop({ reflect: true }) dropdownLabel?: string;
+
+  /** optionally add a calcite-loader component to the control,
+    disabling interaction. with the primary button */
+  @Prop({ reflect: true }) loading?: boolean = false;
+
+  /** is the control disabled  */
+  @Prop({ reflect: true }) disabled?: boolean;
+
+  /** fired when the modal begins the open animation */
+  @Event() primaryButtonClicked: EventEmitter;
+
+  @Watch("color")
+  validateColor() {
+    let color = ["blue", "red", "dark", "light"];
+    if (!color.includes(this.color)) this.color = "blue";
+  }
+
+  @Watch("scale")
+  validateScale() {
+    let scale = ["xs", "s", "m", "l", "xl"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+  }
+
+  @Watch("theme")
+  validateTheme() {
+    let theme = ["dark", "light"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+  }
+
+  @Watch("dropdownIconType")
+  validateDropdownIconType() {
+    let dropdownIconType = ["chevron", "caret"];
+    if (!dropdownIconType.includes(this.dropdownIconType)) this.dropdownIconType = "chevron";
+  }
+
+  connectedCallback() {
+    this.validateColor();
+    this.validateScale();
+    this.validateTheme();
+    this.validateDropdownIconType();
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host dir={dir}>
+        <div>
+          <calcite-button
+            aria-label={this.primaryLabel}
+            color={this.color}
+            scale={this.buttonScale}
+            loading={this.loading}
+            icon={this.primaryIcon}
+            iconPosition="start"
+            disabled={this.disabled}
+            theme={this.theme}
+            onClick={this.primaryButtonClickedHandler}
+          >
+            {this.primaryText}
+          </calcite-button>
+          <div class="split-button__divider-container">
+            <div class="split-button__divider" />
+          </div>
+          <calcite-dropdown
+            alignment="end"
+            dir={dir}
+            theme={this.theme}
+            scale={this.dropdownScale}
+            width={this.dropdownScale}
```

Comment:
should the width be set to the scale here?

Reply:
I thought the `'L'` dropdown scale + width was a bit too big for the current available sizes of this control.  

Originally I'd had `xs` and `xl` versions of the control (to match all available sizes of button), but the proportions between text and icon-only buttons didn't really work on the larger sizes.  So I removed the largest two (which used the `L` scale/width of dropdown), and made the smaller three `s`, `m`, and `l`.

If we were to ever add back these larger sizes, I think using the large dropdown there would make sense.

Reply:
The `width` attribute for dropdown is mutually exclusive from `scale` for dropdown, so those would be two separate pass-through-props I guess? For the scale... we will be updating the spacing on those a bit, but I feel like maybe it's a little weird that the pass-through-prop of `scale="l"` wouldn't result in a child dropdown with a `scale="l"`?